lib/graphql/language/visitor.rb

```diff
@@ -181,14 +181,18 @@ def on_node_with_modifications(node, parent)
             # The user-provided hook requested to remove this node
             new_parent = new_parent && new_parent.delete_child(node)
             return nil, new_parent
+          elsif new_node_and_new_parent.none? { |n| n == nil || n.class < Nodes::AbstractNode }
+            # The user-provided hook returned an array of who-knows-what
+            # ignore the return value and use the original values.
+            return [node, parent]
           else
             new_node_and_new_parent
           end
         else
           # The user-provided hook didn't make any modifications.
           # In fact, the hook might have returned who-knows-what, so
           # ignore the return value and use the original values.
-          new_node_and_new_parent
+          [node, parent]
```

Comment:
Can we keep returning original values here? This is a hot path, so I'd like to avoid making the extra array if we can. (Or, do we need it to make this work?)

```suggestion
          new_node_and_new_parent
```

Reply:
Ah! yes, I didn't realize that everything that called `on_node_with_modifications` was checking if the return value was an array or not.

Would it be more appropriate/performant to return `nil` on line 187 as well?

Reply:
Yes, if it works on line 187, that would be great! I know in _some_ spots we have to make sure to return the right thing, but I haven't dug in for a while. The test suite should cover it!